cpu/instructions.h

```diff
@@ -16,44 +16,53 @@
 //
 static void adc() {
     penaltyop = 1;
-    value = getvalue();
-    result = (uint16_t)a + value + (uint16_t)(status & FLAG_CARRY);
-   
-    carrycalc(result);
-    zerocalc(result);
-    overflowcalc(result, a, value);
-    signcalc(result);
-    
     #ifndef NES_CPU
     if (status & FLAG_DECIMAL) {
         clearcarry();
```

Comment:
That line is a left-over from the bug -- it must be removed (previous carries must be included in additions).

(It's one of two reasons why your use-case still fails.)